src/main/scala/temple/generate/docker/DockerfileGenerator.scala

```diff
@@ -16,10 +16,16 @@ object DockerfileGenerator {
   /** Given a [[temple.generate.docker.ast.Statement]], generate a valid string */
   private def generateStatement(statement: Statement): String =
     statement match {
-      case From(image, tag)        => mkCode("FROM", image, tag.map(":" + _))
-      case Run(executable, params) => mkCode("RUN", buildArrayString(executable +: params))
-      case Cmd(executable, params) => mkCode("CMD", buildArrayString(executable +: params))
-      case Expose(port)            => mkCode("EXPOSE", port.toString)
+      case From(image, tag)               => mkCode("FROM", image, tag.map(":" + _))
+      case Run(executable, params)        => mkCode("RUN", buildArrayString(executable +: params))
+      case Cmd(executable, params)        => mkCode("CMD", buildArrayString(executable +: params))
+      case Expose(port)                   => mkCode("EXPOSE", port.toString)
+      case Env(key, value)                => mkCode("ENV", key, value)
+      case Add(src, dest)                 => mkCode("ADD", src, dest)
+      case Copy(src, dest)                => mkCode("COPY", src, dest)
```

Comment:
Does this work with a space in the file path? Could that ever happen?

Reply:
Well to be a valid file path in the first place, any spaces would need to be escaped, which would get propagated through to the Dockerfile, which also supports escaped spaces in paths.

Reply:
Not sure that’s true, but if you’re sure they’ve already been fully escaped then that’s fine